src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -417,35 +438,141 @@ static MvcCoreLoggerExtensions()
                 LogLevel.Debug,
                 12,
+                "Registered model binder providers, in the following order: {ModelBinderProviders}");
+
+            _attemptingToBindPropertyModel = LoggerMessage.Define<Type, string, Type, string>(
+               LogLevel.Debug,
+               13,
+               "Attempting to bind property '{PropertyContainerType}.{PropertyName}' of type '{ModelType}' using the name '{ModelName}' in request data ...");
+
+            _doneAttemptingToBindPropertyModel = LoggerMessage.Define<Type, string, Type>(
+               LogLevel.Debug,
+               14,
+               "Done attempting to bind property '{PropertyContainerType}.{PropertyName}' of type '{ModelType}'.");
+
+            _foundNoValueForPropertyInRequest = LoggerMessage.Define<string, Type, string, Type>(
+               LogLevel.Debug,
+               15,
+               "Could not find a value in the request with name '{ModelName}' for binding property '{PropertyContainerType}.{ModelFieldName}' of type '{ModelType}'.");
+
+            _foundNoValueInRequest = LoggerMessage.Define<string, string, Type>(
+               LogLevel.Debug,
+               16,
+               "Could not find a value in the request with name '{ModelName}' for binding parameter '{ModelFieldName}' of type '{ModelType}'.");
+
+            _noPublicSettableProperties = LoggerMessage.Define<string, Type>(
+               LogLevel.Debug,
+               17,
+               "Could not bind to model with name '{ModelName}' and type '{ModelType}' as the type has no public settable properties.");
+
+            _cannotBindToComplexType = LoggerMessage.Define<Type>(
+               LogLevel.Debug,
+               18,
+               "Could not bind to model of type '{ModelType}' as there were no values in the request for any of the properties.");
+
+            _cannotBindToFilesCollectionDueToUnsupportedContentType = LoggerMessage.Define<string, Type>(
+               LogLevel.Debug,
+               19,
+               "Could not bind to model with name '{ModelName}' and type '{ModelType}' as the request did not have a content type of either 'application/x-www-form-urlencoded' or 'multipart/form-data'.");
+
+            _cannotCreateHeaderModelBinder = LoggerMessage.Define<Type>(
+               LogLevel.Debug,
+               20,
+               "Could not create a binder for type '{ModelType}' as this binder only supports 'System.String' type or a collection of 'System.String'.");
+
+            _noFilesFoundInRequest = LoggerMessage.Define(
+                LogLevel.Debug,
+                21,
+                "No files found in the request to bind the model to.");
+
+            _attemptingToBindParameter = LoggerMessage.Define<string, string, Type>(
+                LogLevel.Debug,
+                22,
+                "Attempting to bind {ParameterOrProperty} '{ParameterOrPropertyName}' of type '{ModelType}' ...");
```

Comment:
Wondering about using different messages for the parameter and property cases. That way, you'd be able to use `'{PropertyContainerType}.{PropertyName}'` in all messages about properties. Thoughts?